docs/DeveloperGuide.md

```diff
@@ -1,34 +1,243 @@
+ `user` object will be invoked, allowing users to go through a set of instructions to input their `name`, `age`, `gender`, `height`, `weight` and
+ `fitnessLevel`.
+ 
+If a `User` object is instantiated with defined parameters (i.e. `User user = new User(name, age, height, weight
+, gender)`), the values from the given parameters will be stored within the user class without going through
+ `setup()`.
+
+On startup, the `Fitr` class instantiates a `Storage` object and calls its `loadUserProfile()` method. This method
+returns a `User` type object and is referenced by a pre-declared `User` type variable, which is used throughout the
+running session. 
+
+#### 3.2.5 Storage component
+
+Under the storage component, the `StorageManager` class handles the read and write operations of the various list classes available, such as `ExerciseList` and `FoodList`, through classes such as `ExerciseStorage` and `FoodStorage`.
+
+![StorageManager class diagram](images/StorageClassDiagram.png)
+
+_Figure 3: `StorageManager` class diagram_
+
+Figure 3 above shows the associations and dependencies present with the `StorageManager` class.
+
+On startup, the `Fitr` class creates a new `StorageManager` object. The `StorageManager` class will then create the various storage objects required to load that particular file into the program. 
+This will cause the various storage objects to retrieve the text files from the given file paths and loads the data into the program. 
+If the files do not exist, the various storage objects will create the files. For example, if the file that contains the user's exercise data does not exist, the `ExerciseStorage` object will create a new file to store the user's exercise data.
+
+Each time there is a change in the `FoodList`, `ExerciseList` or `User` objects, or when the program is exiting, `writeExerciseList()`, `writeFoodList()` and `writeUserProfile()` methods in `StorageManager` are invoked. 
+This will save the data onto the respective text files on the hard disk.
+
+#### 3.2.6 Common classes
+
+Classes used by multiple components are in the `fitr.common` package.
+
+## 4. Implementation
+
+This section describes how some of the features in Fitr are implemented.
+
+### 4.1 Edit command
+
+When the user enters an edit command, it first passes through `Parser`. Once the input is parsed as an edit command, it is then passed to `EditCommandParser`, where it further parses the user's input. As the user is able to edit either the individual profile characteristics, food or exercise entries, the `EditCommandParser` is able parse what the user intends to edit. For example, if the user intends to edit a food entry, the input is then passed to `EditEntryCommand`, which parses the remaining arguments, and performs the required edit.
+Figure 4 below shows the sequence diagram when the user enters `edit exercise 25/10/2020 1 push ups /100`.
+
+![Sequence diagram for editing exercise](images/EditExerciseSequenceDiagram.png)
```

Comment:
Hi, I've noticed that the activation bar in the Fitr class was not properly closed, despite the edit command being successfully executed. Other than that, the sequence diagram looks good. Keep up the good work! 👍